crates/dojo-core/src/world.cairo

```diff
@@ -263,16 +267,23 @@ mod World {
     ) {
         // Assert can only be called through the executor
         // This is to prevent system from writing to storage directly
-        assert(get_caller_address() == executor_dispatcher::read().contract_address, 'must be called thru executor');
+        assert(
+            get_caller_address() == executor_dispatcher::read().contract_address,
+            'must be called thru executor'
+        );
 
         // Get execution role
         let role = execution_role();
 
-        // Validate the calling system has permission to write to the component
-        assert(
-            is_authorized(context.caller_system, component, AuthRole { id: role }),
-            'system not authorized'
-        );
+        // Validate authorization if role is not set
+        // Otherwise, proceed with the write
+        if role.into() == 0 {
+            // Validate the calling system has permission to write to the component
+            assert(
+                is_authorized(context.caller_system, component, AuthRole { id: role }),
+                'system not authorized'
+            );
+        };
```

Comment:
Authorization flow when role is not set, default resource-scoped role is checked

Reply:
We should make this behavior more explicit in the refactor

Reply:
Got it, will add it to the refactor issue 👍